v2/handler.go

```diff
@@ -0,0 +1,340 @@
+package btclog
+
+import (
+	"context"
+	"fmt"
+	"github.com/btcsuite/btclog"
+	"io"
+	"log/slog"
+	"strconv"
+	"sync"
+	"sync/atomic"
+	"time"
+)
+
+// HandlerOption is the signature of a functional option that can be used to
+// modify the behaviour of the DefaultHandler.
+type HandlerOption func(*handlerOpts)
+
+// handlerOpts holds options that can be modified by a HandlerOption.
+type handlerOpts struct {
+	flag              uint32
+	withTimestamp     bool
+	timeSource        func() time.Time
+	callSiteSkipDepth int
```

Comment:
nit: tho some are self-descriptive, I'd favor putting docs here for future devs.